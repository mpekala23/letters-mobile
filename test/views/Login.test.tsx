import React from "react";
import { LoginScreen } from "@views";
import { login } from "@api";
import { render, fireEvent, toJSON } from "@testing-library/react-native";
import fetchMock from "jest-fetch-mock";

jest.mock("@api", () => ({
  login: jest.fn(),
}));

const setup = (response = {}) => {
  const navigation = { navigate: jest.fn() };
  if (Object.keys(response).length > 0) {
    fetchMock.mockOnce(JSON.stringify(response));
  }
  return render(<LoginScreen navigation={navigation} />);
};

describe("Login screen", () => {
  it("should match snapshot", () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });
  it("should make an api call on login", async () => {
    const { getByText, getByTestId } = setup({
      data: {
        id: "6",
        firstName: "Team",
        lastName: "Ameelio",
        email: "team@ameelio.org",
        phone: "4324324432",
        address1: "Somewhere",
        country: "USA",
        postal: "12345",
        city: "New Haven",
        state: "CT",
      },
      type: "success",
    });
    const loginButton = getByText("Login");
    fireEvent.press(getByText("Login"));
    expect(login).toHaveBeenCalledTimes(1);
  });
});
