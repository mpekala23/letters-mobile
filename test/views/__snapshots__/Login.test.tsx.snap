// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={1}
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ScrollView
          keyboardShouldPersistTaps="handled"
          scrollEnabled={false}
        >
          <View
            style={
              Object {
                "height": 60,
                "width": "100%",
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "paddingHorizontal": 10,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.7}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#3490dc",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 5,
                  },
                  Object {},
                  Object {},
                  Object {},
                ]
              }
              testID="clickable"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                      "fontWeight": "700",
                      "padding": 6,
                    },
                    Object {},
                    Object {},
                    Object {},
                  ]
                }
              >
                Dev Skip
              </Text>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "height": 50,
                  "marginBottom": 10,
                  "width": "100%",
                }
              }
              testID="parent"
            >
              <ScrollView
                keyboardShouldPersistTaps="always"
                scrollEnabled={false}
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "width": "100%",
                    },
                    Object {},
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  placeholder="E-mail Address"
                  rejectResponderTermination={true}
                  secureTextEntry={false}
                  style={
                    Array [
                      Object {
                        "borderColor": "#282828",
                        "borderRadius": 4,
                        "borderWidth": 2,
                        "color": "#282828",
                        "fontSize": 18,
                        "height": 50,
                        "padding": 10,
                        "width": "100%",
                      },
                      Object {},
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <View
                  style={
                    Object {
                      "height": 0,
                      "width": "100%",
                      "zIndex": 5,
                    }
                  }
                >
                  <View />
                </View>
              </ScrollView>
            </View>
            <View
              style={
                Object {
                  "height": 50,
                  "marginBottom": 10,
                  "width": "100%",
                }
              }
              testID="parent"
            >
              <ScrollView
                keyboardShouldPersistTaps="always"
                scrollEnabled={false}
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "width": "100%",
                    },
                    Object {},
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  placeholder="Password"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Array [
                      Object {
                        "borderColor": "#282828",
                        "borderRadius": 4,
                        "borderWidth": 2,
                        "color": "#282828",
                        "fontSize": 18,
                        "height": 50,
                        "padding": 10,
                        "width": "100%",
                      },
                      Object {},
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <View
                  style={
                    Object {
                      "height": 0,
                      "width": "100%",
                      "zIndex": 5,
                    }
                  }
                >
                  <View />
                </View>
              </ScrollView>
            </View>
            <TouchableOpacity
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                }
              }
              activeOpacity={0.2}
              checkedIcon={
                <Text>
                  X
                </Text>
              }
              size={24}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#ededed",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "margin": 5,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "padding": 10,
                  "width": "50%",
                }
              }
              testID="checkbox"
              uncheckedColor="#bfbfbf"
              uncheckedIcon={
                <Text>
                  O
                </Text>
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text>
                  O
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#43484d",
                      "fontWeight": "bold",
                      "marginLeft": 10,
                      "marginRight": 10,
                    }
                  }
                  testID="checkboxTitle"
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  Remember Me
                </Text>
              </View>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "backgroundColor": "#e6e6e6",
                  "height": 1,
                  "marginVertical": 8,
                  "width": "100%",
                }
              }
            />
            <TouchableOpacity
              activeOpacity={0.7}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#3490dc",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 5,
                  },
                  Object {},
                  Object {
                    "width": "100%",
                  },
                  Object {},
                ]
              }
              testID="clickable"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                      "fontWeight": "700",
                      "padding": 6,
                    },
                    Object {},
                    Object {},
                    Object {},
                  ]
                }
              >
                Login
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.7}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "#3490dc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 5,
                  },
                  Object {},
                  Object {
                    "width": "100%",
                  },
                  Object {},
                ]
              }
              testID="clickable"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#3490dc",
                      "fontSize": 20,
                      "fontWeight": "700",
                      "padding": 6,
                    },
                    Object {},
                    Object {},
                    Object {},
                  ]
                }
              >
                Register
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.7}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#3490dc",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 5,
                  },
                  Object {},
                  Object {
                    "backgroundColor": "white",
                    "marginBottom": 30,
                    "width": "60%",
                  },
                  Object {},
                ]
              }
              testID="clickable"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                      "fontWeight": "700",
                      "padding": 6,
                    },
                    Object {},
                    Object {
                      "color": "#3490dc",
                      "fontSize": 16,
                      "fontWeight": "500",
                    },
                    Object {},
                  ]
                }
              >
                Forgot Your Password?
              </Text>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                  }
                }
              >
                By creating an account, you agree to the
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                style={Object {}}
                testID="clickable"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": "400",
                      },
                      Object {
                        "color": "#3490dc",
                        "textDecorationLine": "underline",
                      },
                    ]
                  }
                >
                  Terms of Service
                </Text>
              </TouchableOpacity>
              <Text
                style={
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                  }
                }
              >
                 and 
              </Text>
              <TouchableOpacity
                activeOpacity={0.2}
                style={Object {}}
                testID="clickable"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": "400",
                      },
                      Object {
                        "color": "#3490dc",
                        "textDecorationLine": "underline",
                      },
                    ]
                  }
                >
                  Privacy Policy
                </Text>
              </TouchableOpacity>
              <Text
                style={
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                  }
                }
              >
                .
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 100,
                  "width": "100%",
                }
              }
            />
          </View>
        </ScrollView>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
